Public/Src/Utilities/Native/IO/FileUtilities.cs

```diff
@@ -727,23 +727,16 @@ public static Possible<string> TryGetReparsePointTarget(SafeFileHandle handle, s
         /// </summary>
         public static bool IsDirectorySymlinkOrJunction(string path)
         {
-            try
-            {
-                FileAttributes dirSymlinkOrJunction = FileAttributes.ReparsePoint | FileAttributes.Directory;
-                FileAttributes attributes = FileUtilities.GetFileAttributes(path);
-
-                return (attributes & dirSymlinkOrJunction) == dirSymlinkOrJunction;
-            }
-            catch (NativeWin32Exception)
-            {
-                // FileSystem.Win.
-                return false;
-            }
-            catch (BuildXLException)
+            if (OperatingSystemHelper.IsUnixOS)

```

Comment:
>if (OperatingSystemHelper.IsUnixOS) [](start = 12, length = 35)

Instead of branching, we can add `IsDirectorySymlinkOrJunction` into `IFileSystem`, and then here we simply call `s_fileSystem.IsDirectorySymlinkOrJunction(path)`.

For Win, in `FileSystem.Win.cs`, instead of calling `GetFileAttributes`, you call `TryGetFileAttributes`, which already existed. Thus, you don't need to introduce `TryGetFileAttributes` in this class.

What do you think about that?